src/configureWorkspace/configureDotNetCore.ts

```diff
@@ -249,14 +262,85 @@ function genDockerFile(serviceNameAndRelativePath: string, platform: Platform, o
         .replace(/\$assembly_name\$/g, assemblyNameNoExtension)
         .replace(/\$copy_project_commands\$/g, copyProjectCommands);
 
+    validateForUnresolvedToken(contents);
+
+    return contents;
+}
+
+function validateForUnresolvedToken(contents: string): void {
     let unreplacedToken = extractRegExGroups(contents, /(\$[a-z_]+\$)/, ['']);
     if (unreplacedToken[0]) {
         assert.fail(`Unreplaced template token "${unreplacedToken}"`);
     }
+}
 
-    return contents;
+function generateComposeFiles(dockerfileName: string, platform: Platform, os: PlatformOS | undefined, ports: number[], artifactName: string): ScaffoldFile[] {
+    const serviceName = path.basename(artifactName, path.extname(artifactName)).toLowerCase();
+    const imageName = serviceName;
+    let jsonPorts: string = '';
+    if (ports && ports.length > 0) {
+        const portsArray = ports.map(p => `      - ${p}`).join("\r\n");
```

Comment:
Isn't there a common port mappings generator function?

Reply:
Not that I know of. During launch, it uses the hardcoded value of 80, 443.

Reply:
I mean `getComposePorts` from configure.ts.